### ISSUE TITLE
Dartanalyzer crash during ffi validation

### ISSUE THREAD

**Issue by izolyomi** [dart-lang/sdk#45719]
After upgrading to null-safety with Dart 2.12.x and tested mainly on Linux, `dartanalyzer .` always crashes on our [open-source library](https://github.com/Internet-of-People/iop-dart). Consequently, our pub.dev release never "passes static analysis" on the [score page](https://pub.dev/packages/iop_sdk/versions/5.0.0-alpha1/score)

The crash seems to be caused by an ffi validation bug in dartanalyzer. Here's the resulted stacktrace:
```
$ dartanalyzer .
Analyzing iop-dart...
Unhandled exception:
NoSuchMethodError: The getter 'name' was called on null.
Receiver: null
Tried calling: name
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:54:5)
#1      FfiVerifier._isPointer (package:analyzer/src/generated/ffi_verifier.dart:326:15)
#2      FfiVerifier._isEmptyStruct (package:analyzer/src/generated/ffi_verifier.dart:267:18)
#3      FfiVerifier.visitClassDeclaration (package:analyzer/src/generated/ffi_verifier.dart:66:15)
#4      ClassDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:1536:49)
#5      NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:7611:20)
#6      CompilationUnitImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:2141:21)
#7      RecursiveAstVisitor.visitCompilationUnit (package:analyzer/dart/ast/visitor.dart:731:10)
#8      CompilationUnitImpl.accept (package:analyzer/src/dart/ast/ast.dart:2134:49)
#9      LibraryAnalyzer._computeVerifyErrors (package:analyzer/src/dart/analysis/library_analyzer.dart:419:10)
#10     LibraryAnalyzer.analyzeSync.<anonymous closure> (package:analyzer/src/dart/analysis/library_analyzer.dart:150:7)
#11     _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:397:8)
#12     LibraryAnalyzer.analyzeSync (package:analyzer/src/dart/analysis/library_analyzer.dart:149:11)
#13     LibraryAnalyzer.analyze (package:analyzer/src/dart/analysis/library_analyzer.dart:107:12)
#14     AnalysisDriver._computeAnalysisResult2.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1319:63)
#15     PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:32:15)
#16     AnalysisDriver._computeAnalysisResult2 (package:analyzer/src/dart/analysis/driver.dart:1296:20)
#17     AnalysisDriver._computeAnalysisResult.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1249:14)
#18     _rootRun (dart:async/zone.dart:1354:13)
#19     _CustomZone.run (dart:async/zone.dart:1258:19)
#20     _runZoned (dart:async/zone.dart:1788:10)
#21     runZoned (dart:async/zone.dart:1710:10)
#22     NullSafetyUnderstandingFlag.enableNullSafetyTypes (package:analyzer/dart/element/null_safety_understanding_flag.dart:42:12)
#23     AnalysisDriver._computeAnalysisResult (package:analyzer/src/dart/analysis/driver.dart:1248:40)
#24     AnalysisDriver._computeErrors (package:analyzer/src/dart/analysis/driver.dart:1374:41)
#25     AnalysisDriver.performWork (package:analyzer/src/dart/analysis/driver.dart:981:20)
#26     AnalysisDriverScheduler._run (package:analyzer/src/dart/analysis/driver.dart:2018:24)
<asynchronous suspension>

#0      AnalysisDriver._computeAnalysisResult2.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1349:9)
#1      PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:32:15)
#2      AnalysisDriver._computeAnalysisResult2 (package:analyzer/src/dart/analysis/driver.dart:1296:20)
#3      AnalysisDriver._computeAnalysisResult.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1249:14)
#4      _rootRun (dart:async/zone.dart:1354:13)
#5      _CustomZone.run (dart:async/zone.dart:1258:19)
#6      _runZoned (dart:async/zone.dart:1788:10)
#7      runZoned (dart:async/zone.dart:1710:10)
#8      NullSafetyUnderstandingFlag.enableNullSafetyTypes (package:analyzer/dart/element/null_safety_understanding_flag.dart:42:12)
#9      AnalysisDriver._computeAnalysisResult (package:analyzer/src/dart/analysis/driver.dart:1248:40)
#10     AnalysisDriver._computeErrors (package:analyzer/src/dart/analysis/driver.dart:1374:41)
#11     AnalysisDriver.performWork (package:analyzer/src/dart/analysis/driver.dart:981:20)
#12     AnalysisDriverScheduler._run (package:analyzer/src/dart/analysis/driver.dart:2018:24)
<asynchronous suspension>
```

Strangely, `dart analyze` works just fine, but `dartanalyzer` always crashes:
```
$ dart analyze
Analyzing iop-dart...                  3.3s
No issues found!
```

Though the codebase is not small, it's open source, so I hope this is already enough to reproduce. We might have some ffi-related error, but the analyzer should at least report it instead of just crashing.

**Comment by srawlins** [comment]
This may have been fixed in Dart 2.13. I cannot reproduce with your iot-dart repo using `dart analyze` in Dart 2.13.

**Comment by izolyomi** [comment]
Right, the internal error was fixed, some remaining errors are normally reported by dartanalyzer caused by breaking `ffi` changes coming with 2.13.
